internal/fs/ea_windows.go

```diff
@@ -283,3 +292,34 @@ func setFileEA(handle windows.Handle, iosb *ioStatusBlock, buf *uint8, bufLen ui
 	status = ntStatus(r0)
 	return
 }
+
+// PathSupportsExtendedAttributes returns true if the path supports extended attributes.
+func PathSupportsExtendedAttributes(path string) (bool, error) {
+	var (
+		volumeName      [syscall.MAX_PATH + 1]uint16
+		fsName          [syscall.MAX_PATH + 1]uint16
+		volumeSerial    uint32
+		maxComponentLen uint32
+		fileSystemFlags uint32
+	)
+	utf16Path, err := windows.UTF16PtrFromString(path)
+	if err != nil {
+		return false, err
+	}
+	ret, _, err := procGetVolumeInformationW.Call(
+		uintptr(unsafe.Pointer(utf16Path)),
+		uintptr(unsafe.Pointer(&volumeName[0])),
+		uintptr(len(volumeName)),
+		uintptr(unsafe.Pointer(&volumeSerial)),
+		uintptr(unsafe.Pointer(&maxComponentLen)),
+		uintptr(unsafe.Pointer(&fileSystemFlags)),
+		uintptr(unsafe.Pointer(&fsName[0])),
+		uintptr(len(fsName)),
+	)
+	if ret == 0 {
+		return false, err
+	}
+
+	supportsEAs := (fileSystemFlags & fileSupportsExtendedAttributes) != 0
```

Comment:
Need to confirm if this is the best way to check for extended attributes. Have tested positive and negative scenarios for this check.

Another option would be to check if the file system name is NTFS, which supports Extended Attributes
fsNameStr := syscall.UTF16ToString(fsName[:])
supportsEAs := fsNameStr == "NTFS"